modules/imap/site.js

```diff
@@ -1318,19 +1318,78 @@ var add_email_in_contact_trusted = function(list_email) {
     }
 };
 
+var list_blocked_senders = [];
+get_liste_block_sieve(Hm_Utils.parse_folder_path(hm_list_path()));
+
+function get_liste_block_sieve(detail) {
+    var page = hm_page_name();
+    if (page == 'message_list') {
+        Hm_Ajax.request(
+            [
+                { name: 'hm_ajax_hook', value: 'ajax_liste_block_sieve' },
+                { name: 'imap_server_id', 'value': detail.server_id},
+            ],
+            function (res) {
+                if (res.ajax_liste_block_sieve) {
+                    list_blocked_senders = res.ajax_liste_block_sieve;
+                }
+            }
+        );
+    }
+};
+
 $('.screen-email-unlike').on("click", function() { imap_screen_email(); return false; });
 
 $('.screen-email-like').on("click", function() {
     var list_email = [];
+    var list_msg_uid = [];
+    var email_existing_in_blocked_senders = [];
     $('input[type=checkbox]').each(function() {
         if (this.checked && this.id.search('imap') != -1) {
             let email = $('.'+ this.id +' .from').attr("data-title")
             if (email = email.trim()) {
                 list_email.push(email);
+                if (list_blocked_senders.lenght > 0) {
+                    Object.keys(list_blocked_senders).forEach(key => {
```

Comment:
Since list_blocked_senders is an array (not an object), using Object.keys is unnecessary. Instead, you can iterate directly over the array.
```
list_blocked_senders.forEach((sender, index) => {
        if (sender === email) {
            //your code here
        }
    });
    ```